test/lifecycle_test.dart

```diff
@@ -175,6 +175,29 @@ main() {
 
         sharedTypeTests(testSnapshotType);
       });
+
+      test('triggers error lifecycle events when an error is thrown', () {
+        var mountNode = new DivElement();
+        var renderedInstance = react_dom.render(components2.SetStateTest({}), mountNode);
+        LifecycleTestHelper component = getDartComponent(renderedInstance);
+        LifecycleTestHelper.staticLifecycleCalls.clear();
+        component.setState({"shouldThrow": true});
+
+        expect(component.lifecycleCalls, containsAllInOrder(['getDerivedStateFromError', 'componentDidCatch']));
+        expect(component.state["shouldThrow"], isFalse);
```

Comment:
This would be a good chance to use `reason`, providing more detail on why this is being asserted
```suggestion
        expect(component.state["shouldThrow"], isFalse, 
            reason: 'applies the state returned by `getDerivedStateFromError`');
```